### DISCUSSION TITLE
Not being able to specify self-hosted worker groups is a security issue that shouldn't be paywalled

### DISCUSSION THREAD

**Discussion by Qix-** [community/community#73319] [Actions]
### Select Topic Area

Product Feedback

### Body

I'm working on a fully OSS project that uses self-hosted actions runners to allow testing low-level code on bare metal machines. One of the components of this CI/CD pipeline are actions that must only run on a select few repositories (just one at the moment, maybe more in the future).

The resources they have access to are _real_ machines and custom PCBs/microcontrollers that I want to have some reasonable control over from a security standpoint. It's not a given that every repository will have the same contributors for each, and I want to prevent cross-contamination or accidental execution of unrelated repositories on these very specific actions runners not just for security purposes, but also because they're very constrained, handling only one job at a time, and this is all paid for out of pocket.

The ideal case as it exists now would be to use repository-scoped actions runners, but that's not possible with PAT tokens at the moment. Via PAT tokens, we can only make org-scoped runners - which is a pain, because PATs are the _only_ way to make actions runners work between cold starts, automatically, without manual configuration of each.

With org-scoped runners, the only way to lock them down would be to use worker groups to specify they're only accessible from certain repositories. This means that groups are _the_ way to make workers safe, secure, and manageable for OSS projects that use multple repositories across a Github organization.

They are, however, paywalled to **Enterprise**. This is a **security feature** you're requiring is paid for at a level unobtainable by OSS passion projects. It's not even a feature that requires infrastructure on Github's part (these aren't github-supplied workers). It's clearly an incentive for enterprise signups, which I understand, but it hurts genuine OSS endeavors.

Please reconsider paywalling worker groups for self-hosted workers.

**Comment by onurtufekci34** [comment]
Your feedback and concerns regarding the paywalling of worker groups for self-hosted workers in GitHub are valid, especially in the context of open-source projects and passion projects. This approach can indeed present challenges for OSS endeavors and individual contributors. While I don't have the capability to make changes to GitHub's policies or features, I can offer some suggestions on how to address this situation and potentially advocate for changes:

GitHub Community and Feedback: You can share your feedback and concerns on GitHub's official community forums and feedback channels. Many platform decisions are influenced by user feedback, and this is a way to make your voice heard.

Collaboration: You might consider collaborating with other open-source project maintainers who face similar challenges. Together, you can amplify your concerns and suggestions, which may have a greater impact on GitHub's policy decisions.

Alternative Solutions: While GitHub's built-in features may not fully accommodate your needs, you can explore alternative approaches and tools to help secure and manage your self-hosted runners, even without worker groups. It may involve custom scripts or additional security measures.

Advocacy: Engage with GitHub's open-source and community initiatives to discuss your specific use case and seek support or exemptions. GitHub has been known to support open-source projects in various ways.

Consider Enterprise: Although it might not be ideal, for some projects with the resources, migrating to GitHub Enterprise or seeking GitHub sponsorship for such projects could be an option to access premium features.

Remember that the GitHub platform evolves over time, and feedback from the community can lead to adjustments in policies and features. Your input can contribute to improving the platform's support for open-source projects and passionate individuals who rely on it for their work.

**Reply by Qix-** [reply]
> GitHub Community and Feedback: You can share your feedback and concerns on GitHub's official community forums and feedback channels. Many platform decisions are influenced by user feedback, and this is a way to make your voice heard.

That's exactly where I am. Is this an AI generated response? Do you work for Github?
